modules/http/http-curl-compression.c

```diff
@@ -0,0 +1,34 @@
+/*
```

Comment:
I wonder why are these in the `modules/http/` directory. I do no see any HTTP or CURL specific in the final file, it simply implements different compresion strategies. Shouldn't it go under `lib/` instead? Maybe we could use it in other places, too.

Reply:
This unit depends on zlib, and will eventually depend on additional compression libraries too.
I think it's best to keep them under modules/http until we find another user.

Note that `zlib` was just a transitive dependency of ours. With this PR, it became a direct dep of the HTTP module (`libsyslog-ng' would be too much and unnecessary at the moment).

Reply:
Good point, okay, let's keep it here.

However, I still think that we should not "burn in" that it is http/curl specific, if it is not, maybe we could rename the file and the functions, so we can see that it is general.

Reply:
Are you sure the last 2 comments were intended to go here? This thread is about file/function renames, not compression implementations.

Reply:
You are right. I don't know how they got here :sweat_smile:

Reply:
@therandomstring I still think the compression module is not HTTP or CURL specific. Would you kindly rename the files?

Reply:
Should be done